pkg/cloudprovider/amifamily/resolver.go

```diff
@@ -116,6 +116,15 @@ func New(kubeClient client.Client, ssm ssmiface.SSMAPI, ec2api ec2iface.EC2API,
 	}
 }
 
+func (r Resolver) ResolveAmis(ctx context.Context, provider *v1alpha1.AWS, providerRef *v1alpha5.ProviderRef, instanceTypes []cloudprovider.InstanceType, options *Options) (map[string][]cloudprovider.InstanceType, error) {
```

Comment:
My gut tells me that we should resolve provider/providerRef higher up in the stack.

Reply:
Thinking on this, I'm struggling to see the point of this function since it has the same signature as r.amiProvider.Get. Is there a reason we can't call GetAMIFamily at the callsites?

Reply:
`GetAMIFamily` was not a method on Resolver, I added that and see your point that there is no need for this function. 
Will fix it, thanks!

Reply:
Fixed, this function is removed now.